framework/src/auxkernels/MaterialStdVectorAux.C

```diff
@@ -34,5 +38,9 @@ MaterialStdVectorAux::~MaterialStdVectorAux()
 Real
 MaterialStdVectorAux::getRealValue()
 {
+  if (_selected_qp >= _q_point.size())
+    mooseError("MaterialStdVectorAux.  selected_qp specified as " << _selected_qp << " but there are only " << _q_point.size() << " quadpoints in the element");
+  if (_has_selected_qp)
+    return _prop[_selected_qp][_index];
```

Comment:
Logically the `if (_selected_qp >= _q_point.size())` should be nested inside the `if (_has_selected_qp)`...